test/cpp/end2end/xds/xds_routing_end2end_test.cc

```diff
@@ -96,11 +97,13 @@ TEST_P(LdsDeletionTest, ListenerDeletionIgnored) {
   EXPECT_EQ(response_state->state, AdsServiceImpl::ResponseState::ACKED);
   // Unset LDS resource and wait for client to ACK the update.
   balancer_->ads_service()->UnsetResource(kLdsTypeUrl, kServerName);
-  const auto deadline = absl::Now() + absl::Seconds(30);
+  const auto deadline =
+      absl::Now() + (absl::Seconds(30) * grpc_test_slowdown_factor());
   while (true) {
     ASSERT_LT(absl::Now(), deadline) << "timed out waiting for LDS ACK";
     response_state = balancer_->ads_service()->lds_response_state();
     if (response_state.has_value()) break;
+    absl::SleepFor(absl::Seconds(1) * grpc_test_slowdown_factor());
```

Comment:
uh oh! Why do we need a sleep?

Reply:
The sleep just avoids wasting CPU time.  Without it, this is just a busy loop, constantly checking the same thing over and over again.  With the sleep, we pause to actually give the system time to change state before we check again.

Note that the sleep does not affect the overall 30-second deadline for getting the ACK.

Reply:
something like using a notification might be better